lib/components/sync-status/get-note-titles.ts

```diff
@@ -1,24 +1,25 @@
-import { compact } from 'lodash';
 import noteTitleAndPreview from '../../utils/note-utils';
 
-const getNoteTitles = (ids, notes, limit = Infinity) => {
-  const matchedNotes = ids.map((id, i) => {
-    if (i >= limit) {
-      return;
-    }
+import * as T from '../../types';
 
-    const note = notes.find(thisNote => thisNote.id === id);
-
-    if (!note) {
-      // eslint-disable-next-line no-console
-      console.log(`Could not find note with id '${id}'`);
-      return null;
-    }
+type NoteTitle = {
+  id: String;
+  title: String;
+};
 
-    return { id, title: noteTitleAndPreview(note).title };
-  });
+const getNoteTitles = (
+  ids: String[] = [],
+  notes: T.NoteEntity[] = [],
+  limit: number = Infinity
+) => {
```

Comment:
This is a philosophical question more than an issue but we can add `NoteTitle` as the return type of this function.

On the plus side that enforces the contract such that it can prevent accidentally changing the output type. For instance, as-is type inference determines the return type. We could add `[0]` to the end of the function to return a single result and the inference would auto-update. In a fully-typed codebase we'd catch the errors, but here that will slip by.

If we say it must return `NoteTitle[]` then it will prevent us from unknowingly changing the return value

Reply:
I agree with that, resolved in a0f831f